src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,101 @@
+package duke;
+
+import java.util.Scanner;
+
+import java.io.IOException;
+
+/**
+ * Duke is main driver for storing and displaying Tasks
+ */
+public class Duke {
+    private static final String SAVE_DIRECTORY_PATH = "../data";
+    private static final String SAVE_FILE_PATH = "../data/duke.txt";
+    private static TaskList tasklist = new TaskList();
+    private static Storage storage = new Storage(SAVE_DIRECTORY_PATH, SAVE_FILE_PATH);
+    private static Ui ui = new Ui();
+
+    /**
+     * Greets User, reads user inputs and handles each user command
+     * @param scanner reads user input
+     */
+    private static void run(Scanner scanner) {
+        ui.showGreeting();
+
+        try {
+            storage.checkExistsOrCreateNewFile(tasklist);
+        } catch (Exception e) {
+            System.out.println(e.getMessage());
+        }
+
+        while(scanner.hasNext()) {
+            String inputString = scanner.nextLine();
+            String[] inputs = Parser.splitBySpace(inputString);
+            String input = inputs[0];
+
+            try {
+                if (input.equals("bye")) {
+                    ui.showBye();
+                    break;
+                } else if (input.equals("list")) {
+                    ui.showList(tasklist);
+                } else if (input.equals("mark")) {
+                    ui.showMark(tasklist, inputs[1]);
+                    storage.writeToFile(tasklist.listOfTasksForSaving());
+                } else if (input.equals("unmark")) {
+                    ui.showUnmark(tasklist, inputs[1]);
+                    storage.writeToFile(tasklist.listOfTasksForSaving());
+                } else if (input.equals("todo")) {
+                    if (inputs.length <= 1) {
+                        throw new DukeException("☹ OOPS!!! The description of a todo cannot be empty.");
+                    }
+                    tasklist.appendToDo(inputString);
+                    ui.showAddedTask(tasklist);
+                    storage.writeToFile(tasklist.listOfTasksForSaving());
+                } else if (input.equals("deadline")) {
+                    if (inputs.length <= 1) {
+                        throw new DukeException("☹ OOPS!!! The description of a deadline cannot be empty.");
+                    }
+                    String[] deadlineDescription = Parser.splitBySlash(inputString);
```

Comment:
Should this be plural? Besides, is this just the description of deadline (since the second element is date)? Maybe you can change it to something else?